plugins/modules/gitlab_group_access_token.py

```diff
@@ -0,0 +1,331 @@
+    aliases: ["scope"]
+    choices: ["api", "read_api", "read_registry", "write_registry", "read_repository", "write_repository", "create_runner", "ai_features", "k8s_proxy"]
+  access_level:
+    description:
+      - Access level of the access token.
+    type: str
+    default: maintainer
+    choices: ["guest", "reporter", "developer", "maintainer", "owner"]
+  expires_at:
+    description:
+      - Expiration date of the access token in C(YYYY-MM-DD) format.
+      - Make sure to quote this value in YAML to ensure it is kept as a string and not interpreted as a YAML date.
+    type: str
+    required: true
+  recreate:
+    description:
+      - Whether the access token will be recreated if it already exists.
+      - When V(never) the token will never be recreated.
+      - When V(always) the token will always be recreated.
+      - When V(state_change) the token will be recreated if there is a difference between desired state and actual state.
+    type: str
+    choices: ["never", "always", "state_change"]
+    default: never
+  state:
+    description:
+      - When V(present) the access token will be added to the group if it does not exist.
+      - When V(absent) it will be removed from the group if it exists.
+    default: present
+    type: str
+    choices: [ "present", "absent" ]
+'''
+
+EXAMPLES = r'''
+- name: "Creating a group access token"
+  community.general.gitlab_group_access_token:
+    api_url: https://gitlab.example.com/
+    api_token: "somegitlabapitoken"
+    group: "my_group/my_subgroup"
+    name: "group_token"
+    expires_at: "2024-12-31"
+    access_level: developer
+    scopes:
+      - api
+      - read_api
+      - read_repository
+      - write_repository
+    state: present
+
+- name: "Revoking a group access token"
+  community.general.gitlab_group_access_token:
+    api_url: https://gitlab.example.com/
+    api_token: "somegitlabapitoken"
+    group: "my_group/my_group"
+    name: "group_token"
+    expires_at: "2024-12-31"
+    scopes:
+      - api
+      - read_api
+      - read_repository
+      - write_repository
+    state: absent
+
+- name: "Change (recreate) existing token if its actual state is different than desired state"
+  community.general.gitlab_group_access_token:
+    api_url: https://gitlab.example.com/
+    api_token: "somegitlabapitoken"
+    group: "my_group/my_group"
+    name: "group_token"
+    expires_at: "2024-12-31"
+    scopes:
+      - api
+      - read_api
+      - read_repository
+      - write_repository
+    recreate: state_change
+    state: present
+'''
+
+RETURN = r'''
+result:
+  description: JSON parsed response from the server.
+  returned: always
+  type: dict
+
+error:
+  description: the error message returned by the GitLab API.
+  returned: failed
+  type: str
+  sample: "400: key is already in use"
+
+access_token:
+  description:
+    - API object.
+    - Only contains the value of the token if the token was created or recreated.
+  returned: always
```

Comment:
This is only returned in case of success and `O(state=present)`.
```suggestion
  returned: success and O(state=present)
```